KeyFrame kf
---
KeyFrame nearest_kf
